src/promptflow-core/promptflow/core/_prompty_utils.py

```diff
@@ -264,6 +269,43 @@ def format_stream(llm_response):
     return result
 
 
+def resolve_references(origin):
+    """Resolve all reference in the object."""
+    if isinstance(origin, str):
+        return resolve_reference(origin)
+    elif isinstance(origin, list):
+        return [resolve_references(item) for item in origin]
+    elif isinstance(origin, dict):
+        return {key: resolve_references(value) for key, value in origin.items()}
+    else:
+        return origin
+
+
+def resolve_reference(reference):
+    """
+    Resolve the reference, two types are supported, env, file.
+    When the string format is ${env:ENV_NAME}, the environment variable value will be returned.
+    When the string format is ${file:file_path}, return the loaded json object.
+    """
+    pattern = r"\$\{(\w+):(.*)\}"
+    match = re.match(pattern, reference)
+    if match:
+        reference_type, value = match.groups()
+        if reference_type == "env":
+            return os.environ.get(value)
+        elif reference_type == "file":
+            if not Path(value).exists():
+                raise UserErrorException(f"Cannot find the reference file {value}.")
+            # TODO: support load other type files.
```

Comment:
@wangchao1230 Whether to support loading other types of files?
And whether to support nested references?

Reply:
may support yaml but no need to support nested references

Reply:
offline sync. yaml and json files are supported. Not support nested references.